### PR TITLE
Create RC version of Device Swap

### PR DESCRIPTION
**Pull request by jgarciahospital**
#### What type of PR is this?

* subproject management

#### What this PR does / why we need it:

Create RC version for Device Swap API

#### Special notes for reviewers:

RC version of Device Swap API

### CODE REVIEW

**Comment by bigludo7** [comment]
@jgarciahospital Should we target to provide test definition as well?
I can do them.


-  we release the 0.1-rc
- dev team code them
- using in // the test definition is always very productive for the code
- we collect potentiel bug on the yaml & test definition
- we fix them with a clean v0.1
WDYT?

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml&nocors)
+  - [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml)
+    
+## What's Changed
+### Added
+* Update CODEOWNERS by @ShutingQing in https://github.com/camaraproject/DeviceSwap/pull/2
+* Update README.md w/ Meeting info by @wrathwolf in https://github.com/camaraproject/DeviceSwap/pull/7
+* Update CODEOWNERS by @bigludo7 in https://github.com/camaraproject/DeviceSwap/pull/8
+* Update MAINTAINERS.MD by @bigludo7 in https://github.com/camaraproject/DeviceSwap/pull/9
+* Update CODEOWNERS - adding Jorge Garcia Hospital by @bigludo7 in https://github.com/camaraproject/DeviceSwap/pull/12
```
Comment (hdamker) [inline CHANGELOG.md:34]: I would omit these kind of "changes" from the CHANGELOG, they are internal management changes and have no value for the audience (they still documented in the "full" changelog).

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+* Initial proposal for Device Swap API by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/13
+* Update README.md by @bigludo7 in https://github.com/camaraproject/DeviceSwap/pull/15
+* Clean error and description by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/18
+* Update device-swap description by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/17
+* Update APi actions descriptions by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/21
```
Comment (hdamker) [inline CHANGELOG.md:39]:
```suggestion
* Initial proposal for Device Swap API by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/13
* Clean error and description by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/18
* Updated device-swap description by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/17
* Updated API actions descriptions by @jgarciahospital in https://github.com/camaraproject/DeviceSwap/pull/21
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+### Fixed
+N/A
+### Removed
+N/A
+
+## New Contributors
+* @ShutingQing made their first contribution in https://github.com/camaraproject/DeviceSwap/pull/2
+* @wrathwolf made their first contribution in https://github.com/camaraproject/DeviceSwap/pull/7
+* @bigludo7 made their first contribution in https://github.com/camaraproject/DeviceSwap/pull/8
+* @jgarciahospital made their first contribution in https://github.com/camaraproject/DeviceSwap/pull/13
```
Comment (hdamker) [inline CHANGELOG.md:52]:
```suggestion
* @bigludo7 made their first contribution in https://github.com/camaraproject/DeviceSwap/pull/8
* @jgarciahospital made their first contribution in https://github.com/camaraproject/DeviceSwap/pull/13
```

File: README.md [hunk 4]
```diff
@@ -26,3 +26,16 @@ Repository to describe, develop, document and test the Device Swap API family
     <!-- Note: the $sub-project-mailinglistname$ is either already existing or will be created by the CAMARA Admin Team  -->
     * Subscribe / Unsubscribe to the mailing list of this Sub Project <https://lists.camaraproject.org/g/sp-dsw>.
     * A message to the community of this Sub Project can be sent using <sp-dsw@lists.camaraproject.org>.
+
+## Status and released versions
```
Comment (hdamker) [inline README.md:30]: This sections between "Scope" and "Contributing", title in the template is "Release Information", cf. https://github.com/camaraproject/Template_API_Repository/blob/main/README.md

File: README.md [hunk 5]
```diff
@@ -26,3 +26,16 @@ Repository to describe, develop, document and test the Device Swap API family
...
+* The pre-release r1.1 contains the version 0.1.0-rc.1 of the Device Swap API. It is the first release candidate of the v0.1.0. The release tag is [r1.1](https://github.com/camaraproject/DeviceSwap/tree/r1.1).
+- 0.1.0-rc.1 Device Swap API definition **with inline documentation**:
+  - OpenAPI [YAML spec file](https://github.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml)
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml&nocors)
+  - [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml)
+
+## Contributorship and mailing list
+* To subscribe / unsubscribe to the mailing list of this Sub Project and thus be / resign as Contributor please visit <https://lists.camaraproject.org/g/sp-pdd>.
+* A message to all Contributors of this Sub Project can be sent using <sp-pdd@lists.camaraproject.org>.
```
Comment (hdamker) [inline README.md:41]: These lines are already there in 26-28

**Comment by bigludo7** [comment]
+ Add the Test Definition in the changelog
Oct15th meeting discussion

File: CHANGELOG.md (hunk 1 above)
Comment (jgarciahospital) [reply CHANGELOG.md]: solved

File: CHANGELOG.md (hunk 2 above)
Comment (jgarciahospital) [reply CHANGELOG.md]: included as suggested

File: CHANGELOG.md (hunk 3 above)
Comment (jgarciahospital) [reply CHANGELOG.md]: also cleaned

File: README.md (hunk 4 above)
Comment (jgarciahospital) [reply README.md]: Moved and title updated

File: README.md (hunk 5 above)
Comment (jgarciahospital) [reply README.md]: removed

**Comment by jgarciahospital** [comment]
Thanks @hdamker for the review,  hope all the proposals are now resolved. Checklist is also already included.

@bigludo7 both user stories and test definitions are included now in the changelog, as agreed to merge r1.1 with all the required information.

**Review by bigludo7 — APPROVED** [review]
LGTM to me
I add the Release Management team as reviewer

**Comment by jgarciahospital** [comment]
@hdamker @tanjadegroot could you please validate latest changes?

File: documentation/SupportingDocuments/device-swap-API-Readiness-Checklist.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# API Readiness Checklist
+
+Checklist for device-swap 0.1.0-rc in r1.1
```
Comment (hdamker) [inline documentation/SupportingDocuments/device-swap-API-Readiness-Checklist.md:3]:
```suggestion
Checklist for device-swap 0.1.0-rc.1 in r1.1
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+# r1.1
+## Release Notes
+
+This release contains the definition and documentation of
+* device-swap v0.1.0-rc.1
+
+The API definition(s) are based on
+* Commonalities v0.4.0
+* Identity and Consent Management v0.2.0
```
Comment (hdamker) [inline CHANGELOG.md:17]:
```suggestion
* Identity and Consent Management v0.2.1
```
(no API relevant changes, just the latest version)

File: README.md [hunk 8]
```diff
@@ -15,7 +15,15 @@ Repository to describe, develop, document and test the Device Swap API family
...
-* Location: virtually  
+* Location: virtually
+
+## Release Information
+* Note: Please be aware that the project will have frequent updates to the main branch. There are no compatibility guarantees associated with code in any branch, including main, until a new release is created. For example, changes may be reverted before a release is created. **For best results, use the latest available release**.
+* The pre-release r1.1 contains the version 0.1.0-rc.1 of the Device Swap API. It is the first release candidate of the v0.1.0. The release tag is [r1.1](https://github.com/camaraproject/DeviceSwap/tree/r1.1).
+- 0.1.0-rc.1 Device Swap API definition **with inline documentation**:
+  - OpenAPI [YAML spec file](https://github.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml)
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml&nocors)
+  - [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/blob/r1.1/code/API_definitions/device-swap.yaml)
```
Comment (hdamker) [inline README.md:26]:
```suggestion
  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/r1.1/code/API_definitions/device-swap.yaml&nocors)
  - [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/DeviceSwap/r1.1/code/API_definitions/device-swap.yaml)
```

File: documentation/SupportingDocuments/device-swap-API-Readiness-Checklist.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+| Nr | API release assets  | alpha | release-candidate |  initial<br>public | stable<br> public | Status | Comments |
+|----|----------------------------------------------|:-----:|:-----------------:|:-------:|:------:|:----:|:----:|
+|  1 | API definition                               |   M   |         M         |    M    |    M   |  Y   | [link](code/API_definitions/device-swap.yaml) |
```
Comment (hdamker) [inline documentation/SupportingDocuments/device-swap-API-Readiness-Checklist.md:7]:
```suggestion
|  1 | API definition                               |   M   |         M         |    M    |    M   |  Y   | [link](/code/API_definitions/device-swap.yaml) |
```

File: documentation/SupportingDocuments/device-swap-API-Readiness-Checklist.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y  | Embed documentation into API spec - [link](/code/API_definitions/device-swap.yaml) |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   Y  | [check](/documentation/API_documentation/device-swap_check_User_Story.md) [retrieve](/documentation/API_documentation/device-swap_retrieve_User_Story.md) |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y  | [check](/code/Test_definitions/device-swap-check.feature) [retrieve](/code/Test_definitions/device-swap-retrieve-date.feature) |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   Y  | [check](/code/Test_definitions/device-swap-check.feature) [retrieve](/code/Test_definitions/device-swap-retrieve-date.feature) |
+|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   N  |  Fall24 EXCEPTION: Test results not available (*)    |
```
Comment (hdamker) [inline documentation/SupportingDocuments/device-swap-API-Readiness-Checklist.md:15]: Not part of Fall24, and not relevant for release candidate
```suggestion
|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   N  |      |
```

**Review by hdamker — CHANGES_REQUESTED** [review]
* Review issue created: https://github.com/camaraproject/ReleaseManagement/issues/114
* Some smaller, formal changes suggested
* Please create a release tracker by using the button on https://lf-camaraproject.atlassian.net/wiki/x/aD3e

**Review by hdamker — APPROVED** [review]
LGTM from Release Management perspective - please create the release.

**Comment by hdamker** [comment]
> https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/32014423/device-swap+v0.1.0

Done minor edits on the page ... for example is the M3 date not applicable for this release outside the release cycle.

**Comment by hdamker** [comment]
> PS. we shall avoid in next meta-release to include minor/patch version of commonalities and I&CM...

Good point, we have identified this also in Release Management. The minor (and pre-release) versions are relevant for the release management, but probably don't belong into the API definition at all. The detailed version will most probably go (only?) into the API-readiness-checklist.

@cc: @tanjadegroot

**Pull request merged**